posthog/tasks/tasks.py

```diff
@@ -40,11 +41,14 @@ def redis_heartbeat() -> None:
     autoretry_for=(
         # Important: Only retry for things that might be okay on the next try
         CHQueryErrorTooManySimultaneousQueries,
+        CeleryConcurrencyLimitExceeded,
     ),
-    retry_backoff=1,
-    retry_backoff_max=2,
-    max_retries=3,
+    retry_backoff=True,
+    retry_backoff_max=10,
+    max_retries=20,
 )
+@limit_concurrency(70)  # Do not go above what CH can handle
```

Comment:
This should probably be more? maybe 90? I think at some point we'll end up with most queries on the frontend being process_query_tasks